### PR TITLE
#4 Virtual Machine Resource

### PR DESCRIPTION
**Pull request by ja6a**
Provide the ability to create a skytap virtual machine. The resource adds an existing VM from an existing template. Only one can be added at a time. The VM will be automatically switched to a running state. Only the name is able to be changed at this stage.
E.g. 
```
resource "skytap_environment" "foo" {
 		template_id = "1448141"
 		name 		= "my env"
 		description = "This is my environment"
 	}

 	resource "skytap_vm" "bar" {
		environment_id    = "${skytap_environment.foo.id}"
   		template_id       = "1448141"
 		vm_id      	  = "36628546"
 	}
```

### CODE REVIEW

File: skytap/config.go [hunk 1]
```diff
@@ -21,6 +21,7 @@ type SkytapClient struct {
 	environmentsClient skytap.EnvironmentsService
 	templatesClient    skytap.TemplatesService
 	networksClient     skytap.NetworksService
+	VMsClient          skytap.VMsService
```
Comment (pegerto) [inline skytap/config.go]: probably vmsClient, help readability.

File: skytap/resource_skytap_vm.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+			},
+
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.NoZeroValues,
+				Computed:     true,
+			},
+
+			"runstate": {
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: the user should not configure runstate.

File: skytap/resource_skytap_vm.go [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+			VMID:       []string{templateVMID},
+		}
+
+		log.Printf("[DEBUG] vm create options: %#v", createOpts)
+		environment, err := client.Create(ctx, environmentID, &createOpts)
+		if err != nil || len(environment.VMs) == 0 {
+			return errors.Errorf("error creating vm: %v", err)
+		}
+
+		vm := mostRecentVM(environment.VMs)
```
Comment (pegerto) [inline skytap/resource_skytap_vm.go]: document why the mostRecentVM, in comments, and in a github issue.

File: skytap/config.go (hunk 1 above)
Comment (ja6a) [reply skytap/config.go]: done

File: skytap/resource_skytap_vm.go (hunk 2 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go]: done

File: skytap/resource_skytap_vm.go (hunk 3 above)
Comment (ja6a) [reply skytap/resource_skytap_vm.go]: will raise a new issue.

**Pull request merged**